crypto/evp/digest.c

```diff
@@ -162,7 +162,8 @@ int EVP_DigestFinal_ex(EVP_MD_CTX *ctx, unsigned char *md, unsigned int *size)
 {
     int ret;
 
-    OPENSSL_assert(ctx->digest->md_size <= EVP_MAX_MD_SIZE);
+    if (!ossl_assert(ctx->digest->md_size <= EVP_MAX_MD_SIZE))
+        return 0;
```

Comment:
Here is another angle. Earlier I mentioned "unjustified OPENSSL_asserts". But which are those? In first hand those that involve treatment of input provided by caller. From this viewpoint OPENSSL_assert is actually ... justified, but it still makes sense to get rid of it, because it's actually excessive. But does it actually make sense to replace it with ossl_assert? Not really, because ctx->digest->md_size is supposed to be constant. This assert is meaningful when you develop corresponding EVP "provider" module, but otherwise the condition is *always* met. In other words it makes lesser sense to replace it with if(ossl_assert), plain assert alone would be sufficient. [Unless of course one starts arguing that failure to meet the condition can actually be considered as indication of memory corruption, in which case OPENSSL_assert would be actually more appropriate.]